File: ClientSide/src/app/DTOs/Post/ShowPostDTO.ts
import { CommentDTO } from '../CommentDTOs/CommentDTO';
import { UserDTO } from '../Account/UserDTO';
import { LikeDTO } from './LikeDTO';
export interface ShowPostDTO {
    id:number,
    postText: string;
    fileName: string;
    userId: number;
    canalId: number;
    dateTime : Date;
    comments: CommentDTO[],
    likes : LikeDTO[],
    user:UserDTO
}

File: BBKBootcampSocial.Core/DTOs/Post/ShowPostDTO.cs
using BBKBootcampSocial.Core.DTOs.Account;
using BBKBootcampSocial.Core.DTOs.Comment;
using System;
using System.Collections.Generic;

namespace BBKBootcampSocial.Core.DTOs.Post
{
    public class ShowPostDTO
    {
        public string PostText { get; set; }
        public string FileName { get; set; }
        public long UserId { get; set; }
        public long? CanalId { get; set; }
        public DateTime DateTime { get; set; }
        public long? ParentId { get; set; }
        public long Id { get; set; }


        public LoginUserInfoDTO User { get; set; }

        public IEnumerable<CommentDTO> Comments{ get; set; }
        public IEnumerable<LikeDTO> Likes{ get; set; }
    }
}
